stan/math/rev/core/operator_logical_or.hpp

```diff
@@ -10,39 +11,48 @@ namespace math {
  * Return the logical disjunction of the values of the two
  * arguments as defined by <code>||</code>.
  *
+ * @tparam Var1 value type of a var
+ * @tparam Var2 value type of a var
  * @param[in] x first argument
  * @param[in] y second argument
  * @return disjuntion of the argument's values
  */
-inline bool operator||(const var& x, const var& y) {
+template <typename Var1, typename Var2, require_all_var_t<Var1, Var2>...>
+inline bool operator||(Var1&& x, Var2&& y) {
```

Comment:
[optional]
The boolean ops can also be generalized in `prim` for all autodiff types now.